src/redux/actions/pipeline/runQC.js

```diff
@@ -42,12 +56,27 @@ const runQC = (experimentId) => async (dispatch, getState) => {
   const { processing } = getState().experimentSettings;
   const { changedQCFilters } = processing.meta;
 
-  if (changedQCFilters.size === 1 && changedQCFilters.has('configureEmbedding')) {
-    runOnlyConfigureEmbedding(
-      experimentId,
-      processing.configureEmbedding.embeddingSettings.method,
-      dispatch,
-    );
+  const embeddingChanged = changedQCFilters.has('embeddingSettings');
+  const clusteringChanged = changedQCFilters.has('clusteringSettings');
+  const otherChanged = [...changedQCFilters].some((value) => value !== 'embeddingSettings' && value !== 'clusteringSettings');
+
+  // if only embedding or clustering changed
+  if (!otherChanged) {
+    await dispatch(saveProcessingSettings(experimentId, 'configureEmbedding'));
```

Comment:
```suggestion
const embeddingChanged = changedQCFilters.has('embeddingSettings');
const clusteringChanged = changedQCFilters.has('clusteringSettings');

if (embeddingChanged || clusteringChanged) {
  await dispatch(saveProcessingSettings(experimentId, 'configureEmbedding'));

```

Reply:
So now that means that if there's more than one changed filter, running it with changed embedding or clustering in the last step will discard them all and run only embedding / clustering?

Reply:
Also if you move back the `await dispatch(saveProcessingSettings(experimentId, 'configureEmbedding'));` line to the `runOnlyClustering` and `runOnlyConfigureEmbedding` functions, this whole block can be only:

```
  const embeddingChanged = changedQCFilters.has('embeddingSettings');
  const clusteringChanged = changedQCFilters.has('clusteringSettings');

    if (embeddingChanged) {
      runOnlyConfigureEmbedding(
        experimentId,
        processing.configureEmbedding.embeddingSettings.method,
        dispatch,
      );
    }
    if (clusteringChanged) {
      runOnlyClustering(
        experimentId,
        processing.configureEmbedding.clusteringSettings.methodSettings.louvain.resolution,
        dispatch,
      );
    }
```

Reply:
> So now that means that if there's more than one changed filter, running it with changed embedding or clustering in the last step will discard them all and run only embedding / clustering?

The problem with this is that it will run the conditional block if there are other changes to Data Processing in addition to an embedding or clustering change. I want the condition to only run if there is **only** an embedding or clustering change.

Reply:
> Also if you move back the `await dispatch(saveProcessingSettings(experimentId, 'configureEmbedding'));` line to the `runOnlyClustering` and `runOnlyConfigureEmbedding` functions, this whole block can be only:
> 
> ```
>   const embeddingChanged = changedQCFilters.has('embeddingSettings');
>   const clusteringChanged = changedQCFilters.has('clusteringSettings');
> 
>     if (embeddingChanged) {
>       runOnlyConfigureEmbedding(
>         experimentId,
>         processing.configureEmbedding.embeddingSettings.method,
>         dispatch,
>       );
>     }
>     if (clusteringChanged) {
>       runOnlyClustering(
>         experimentId,
>         processing.configureEmbedding.clusteringSettings.methodSettings.louvain.resolution,
>         dispatch,
>       );
>     }
> ```

happy to move the line back if you have a preference. I moved it out so that it wasn't duplicated between the two functions

Reply:
I was only thinking in case we want to use the `runOnlyClustering` and `runOnlyConfigureEmbedding` functions  somewhere else in the code, someone might expect them to do a save and forget to `dispatch(save` before running them.